packages/fetch-extension/src/components/chat/chat-init-popup.tsx

```diff
@@ -3,136 +3,129 @@ import { PrivacySetting } from "@keplr-wallet/background/build/messaging/types";
+  }) => {
+    const { chainStore, accountStore, analyticsStore, chatStore } = useStore();
 
-  const { chainStore, accountStore, analyticsStore } = useStore();
-  const current = chainStore.current;
-  const walletAddress = accountStore.getAccount(
-    chainStore.current.chainId
-  ).bech32Address;
+    const userState = chatStore.userDetailsStore;
 
-  const navigate = useNavigate();
-  // address book values
+    const current = chainStore.current;
+    const walletAddress = accountStore.getAccount(
+      chainStore.current.chainId
+    ).bech32Address;
+    const navigate = useNavigate();
+    // address book values
 
-  const [selectedPrivacySetting, setSelectedPrivacySetting] =
-    useState<PrivacySetting>(
-      userState?.messagingPubKey.privacySetting
-        ? userState?.messagingPubKey.privacySetting
-        : PrivacySetting.Everybody
-    );
+    const [selectedPrivacySetting, setSelectedPrivacySetting] =
+      useState<PrivacySetting>(
+        userState.messagingPubKey.privacySetting
+          ? userState.messagingPubKey.privacySetting
+          : PrivacySetting.Everybody
+      );
+    const requester = new InExtensionMessageRequester();
 
-  const requester = new InExtensionMessageRequester();
+    const registerAndSetMessagePubKey = async () => {
+      setLoadingChats(true);
+      try {
+        const messagingPubKey = await requester.sendMessage(
+          BACKGROUND_PORT,
+          new RegisterPublicKey(
+            GRAPHQL_URL.MESSAGING_SERVER,
+            current.chainId,
+            userState.accessToken,
+            walletAddress,
+            selectedPrivacySetting
+          )
+        );
+        chatStore.userDetailsStore.setMessagingPubKey(messagingPubKey);
 
-  const registerAndSetMessagePubKey = async () => {
-    setLoadingChats(true);
-    try {
-      const messagingPubKey = await requester.sendMessage(
-        BACKGROUND_PORT,
-        new RegisterPublicKey(
-          GRAPHQL_URL.MESSAGING_SERVER,
-          current.chainId,
-          userState.accessToken,
-          walletAddress,
-          selectedPrivacySetting
-        )
-      );
-      store.dispatch(setMessagingPubKey(messagingPubKey));
-      navigate("/chat", { replace: true });
-    } catch (e) {
-      // Show error toaster
-      console.error("error", e);
-      store.dispatch(resetUser({}));
-      // Redirect to home
-      navigate("/", { replace: true });
-    } finally {
-      analyticsStore.logEvent("chat_privacy_click", {
-        action: selectedPrivacySetting,
-      });
-      setIsOpendialog(false);
-      setLoadingChats(false);
-    }
-  };
+        navigate("/chat", { replace: true });
+      } catch (e) {
+        // Show error toaster
+        console.error("error", e);
+        chatStore.userDetailsStore.resetUser();
+        // Redirect to home
+        navigate("/", { replace: true });
+      } finally {
+        analyticsStore.logEvent("chat_privacy_click", {
+          selectedPrivacySetting,
```

Comment:
@AbbasAliLokhandwala action property is missing in the event

```
analyticsStore.logEvent("chat_privacy_click", {
        action: selectedPrivacySetting,
      });
```

Reply:
Except this everything looks good. Also ask gaurav to revalidate both store migration and analytics